### PR TITLE
os/board: add esp32 lcd driver

### PR DESCRIPTION
**Pull request by SRCN-LiQin**
add lcd driver of esp32 wrover kit, base on the
fb/lcd driver structure.
@gcjjyy

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [c77c4a88d22783f0b1a4c418de9c852d9dd6b25b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c77c4a88d22783f0b1a4c418de9c852d9dd6b25b] - Code Rule Check Result:
os/board/esp32_devkit/src/esp32_lcd.c:110: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/board/esp32_devkit/src/esp32_lcd.c:110: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
os/board/esp32_devkit/src/esp32_lcd.c:110: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxV)
os/board/esp32_devkit/src/esp32_lcd.c:194: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/board/esp32_devkit/src/esp32_lcd.c:470: ERROR: [SPC_M_OPR] spaces required around that '?' (ctx:VxW)

File: os/arch/xtensa/src/esp32/esp32_gpio.c [hunk 1]
```diff
@@ -203,7 +203,7 @@ int esp32_configgpio(int pin, gpio_pinattr_t attr)
 
 	/* Handle output pins */
 
-	else if ((attr & OUTPUT) != 0) {
+	if ((attr & OUTPUT) != 0) {
```
[the remark this answers is not in the document]
Comment (SRCN-LiQin) [reply os/arch/xtensa/src/esp32/esp32_gpio.c]: the gpio direction maybe set to both in&out , with current logic, it would be only one direction.

**Comment by seinfra** [comment]
Target : [4df4e4ee715a358ee5ef9aae774d75c692fc635b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4df4e4ee715a358ee5ef9aae774d75c692fc635b] - Code Rule Check (C++) OK.

File: os/arch/xtensa/src/esp32/esp32_gpio.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (SRCN-LiQin) [reply os/arch/xtensa/src/esp32/esp32_gpio.c]: this modification should not be included here, removed, thanks

**Comment by seinfra** [comment]
Target : [0267b7a4096aa4f6f999b007eb38fddbf2c15ecb] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0267b7a4096aa4f6f999b007eb38fddbf2c15ecb] - Code Rule Check OK.

File: os/arch/xtensa/src/esp32/Kconfig [hunk 2]
```diff
@@ -11,6 +11,10 @@ config ESP32_UART
 	bool
 	default n
 
+config ESP32_SPI
```
Comment (xixidodo) [inline os/arch/xtensa/src/esp32/Kconfig]: looks good to me.

**Pull request merged**